diff --git a/cups/http.c b/cups/http.c
index c7e6e3b..365883f 100644
--- a/cups/http.c
+++ b/cups/http.c
@@ -3587,7 +3587,7 @@ http_add_field(http_t       *http,	/* I - HTTP connection */
                const char   *value,	/* I - Value string */
                int          append)	/* I - Append value? */
 {
-  char		temp[1024],		/* Temporary value string */
+  char		temp[1024] = {0},		/* Temporary value string */
 		combined[HTTP_MAX_VALUE];
 					/* Combined value string */
   size_t	fieldlen,		/* Length of existing value */
