diff --git a/expat/lib/xmlparse.c b/expat/lib/xmlparse.c
index d44baa6..d0252b1 100644
--- a/expat/lib/xmlparse.c
+++ b/expat/lib/xmlparse.c
@@ -1038,11 +1038,15 @@ callProcessor(XML_Parser parser, const char *start, const char *end,
   return ret;
 }
 
+void __msan_unpoison(void *, size_t);
+
 static XML_Bool /* only valid for root parser */
 startParsing(XML_Parser parser) {
   /* hash functions must be initialized before setContext() is called */
-  if (parser->m_hash_secret_salt == 0)
+  if (parser->m_hash_secret_salt == 0) {
     parser->m_hash_secret_salt = generate_hash_secret_salt(parser);
+    __msan_unpoison(&parser->m_hash_secret_salt, sizeof(parser->m_hash_secret_salt));
+  }
   if (parser->m_ns) {
     /* implicit context only set for root parser, since child
        parsers (i.e. external entity parsers) will inherit it
