diff --git a/nss/lib/softoken/pkcs11.c b/nss/lib/softoken/pkcs11.c
index 9c0d93e..3dec902 100644
--- a/nss/lib/softoken/pkcs11.c
+++ b/nss/lib/softoken/pkcs11.c
@@ -5165,7 +5165,7 @@ sftk_searchDatabase(SFTKDBHandle *handle, SFTKSearchResults *search,
     CK_RV crv;
     int objectListSize = search->array_size - search->size;
     CK_OBJECT_HANDLE *array = &search->handles[search->size];
-    SDBFind *find;
+    SDBFind *find = NULL;
     CK_ULONG count;
 
     crv = sftkdb_FindObjectsInit(handle, pTemplate, ulCount, &find);
