diff --git a/nss/lib/freebl/freebl.gyp b/nss/lib/freebl/freebl.gyp
index 14f213c..c4beb34 100644
--- a/nss/lib/freebl/freebl.gyp
+++ b/nss/lib/freebl/freebl.gyp
@@ -16,19 +16,6 @@
       'dependencies': [
         '<(DEPTH)/exports.gyp:nss_exports'
       ],
-      'conditions': [
-        [ 'cc_is_clang==1 and force_integrated_as!=1', {
-          'cflags': [
-            '-no-integrated-as',
-          ],
-          'cflags_mozilla': [
-            '-no-integrated-as',
-          ],
-          'asflags_mozilla': [
-            '-no-integrated-as',
-          ],
-        }],
-      ],
     },
     {
       'target_name': 'intel-gcm-wrap_c_lib',
diff --git a/nss/lib/freebl/freebl_base.gypi b/nss/lib/freebl/freebl_base.gypi
index 8ef46da..24a21a8 100644
--- a/nss/lib/freebl/freebl_base.gypi
+++ b/nss/lib/freebl/freebl_base.gypi
@@ -90,19 +90,6 @@
             'mpi/mpi_amd64_common.S',
             'mpi/mp_comba.c',
           ],
-          'conditions': [
-            [ 'cc_is_clang==1 and fuzz!=1 and coverage!=1 and force_integrated_as!=1', {
-              'cflags': [
-                '-no-integrated-as',
-              ],
-              'cflags_mozilla': [
-                '-no-integrated-as',
-              ],
-              'asflags_mozilla': [
-                '-no-integrated-as',
-              ],
-            }],
-          ],
         }],
         [ 'target_arch=="ia32"', {
           'sources': [
